MIGRATION.md

```diff
@@ -0,0 +1,91 @@
+# Migrating to the `cloudant-python-sdk` library 
+This document is to assist in migrating from the legacy `python-cloudant` to the newly supported [`cloudant-python-sdk`](https://github.com/IBM/cloudant-python-sdk).
+
+## Initializing the client connection
+There are several ways to create a client connection in `cloudant-python-sdk`:
+1. [Enviornment variables](https://github.com/IBM/cloudant-python-sdk#authentication-with-environment-variables)
+2. [External configuration file](https://github.com/IBM/cloudant-python-sdk#authentication-with-external-configuration)
+3. [Programmatically](https://github.com/IBM/cloudant-python-sdk#programmatic-authentication)
+
+[See the README](https://github.com/IBM/cloudant-python-sdk#code-examples) for code examples on using enviornment variables.
+
+## Other differences
+1. The `cloudant-python-sdk` library does not support local dictionary caching of database and document objects.
+1. There's no context manager feature in `cloudant-python-sdk`.  You'll need to create the client, connect to the database, fetch and save a document as separate commands.
```

Comment:
Ok right and there's multiple context managers within the library e.g. one for creating a Cloudant session, another for handling documents, one for QueryResult, etc
What about something like:
```There's no context manager feature in `cloudant-python-sdk`.  All necessary setup and teardown will need to be added by the user when working with the libraries API operations.```

Reply:
oh, I see! that connect to db part related to a session context manager rather than a doc one.
>All necessary setup and teardown

this sounds more onerous than it really is, because it's basically automatic - going to have a think!

Reply:
It depends on how you connect to the client.  You can use the `connect` parameter during client instantiation or create the client and then do `client.connect()`.  You'll also want to disconnect and end the auth session with `client.disconnect()`.

Reply:
I was meaning the client setup and teardown is basically automatic in cloudant-python-sdk, whereas saying the user needs to add setup and teardown gives the impression that it isn't. How about something like:
```markdown
There are no context managers in `cloudant-python-sdk`. To reproduce
the behaviour of the `python-cloudant` context managers in `cloudant-python-sdk`
users need to explicitly call the specific operations against the remote HTTP
API. For example, in the case of the document context manager, this would mean
doing both a `get_document` to fetch and a `put_document` to save.
```

Reply:
Right, this sounds good to me!